packages/maskbook/src/plugins/BarnBridge/UI/SmartYieldPoolsView.tsx

```diff
@@ -0,0 +1,170 @@
+    }, [syData, syPortfolioData])
+
+    if (_syDataState.length <= 1 || loadingPools || loadingPortfolio) {
+        return <CircularProgress className={classes.progress} size={50} />
+    }
+
+    const protocols = Object.keys(_syDataState[0])
+    //#endregion
+
+    //#region tabs
+    // Tabs Construction
+    const SYPoolsTab = (props: SmartYieldPoolsViewProps) => (
+        <Tab
+            {...props}
+            className={classes.tab}
+            key="SYMarkets"
+            label={
+                <div>
+                    <BarnBridgeSmartYieldBarChart className={classes.icon} style={{ verticalAlign: 'middle' }} />
+                    {t('plugin_barnbridge_sy_markets')}
+                </div>
+            }
+        />
+    )
+
+    const SYWalletTab = (props: SmartYieldPoolsViewProps) => (
+        <Tab
+            {...props}
+            className={classes.tab}
+            key="SYPortfolio"
+            label={
+                <div>
+                    <BarnBridgeSmartYieldWallet className={classes.icon} style={{ verticalAlign: 'middle' }} />
+                    {t('plugin_barnbridge_sy_portfolio')}
+                </div>
+            }
+        />
+    )
+
+    const tabs = [<SYPoolsTab />, <SYWalletTab />]
+    //#endregion
+
+    return (
+        <CardContent className={classes.root}>
+            <Tabs
+                className={classes.tabs}
+                classes={{
+                    indicator: classes.indicator,
+                }}
+                indicatorColor="secondary"
+                textColor="inherit"
+                variant="fullWidth"
+                value={tabIndex}
+                orientation="horizontal"
+                onChange={(ev: React.ChangeEvent<{}>, newValue: number) => {
+                    setTabIndex(newValue)
+                }}>
+                {tabs}
+            </Tabs>
+            <Paper className={classes.paper}>
+                {tabIndex === 0 ? GenerateSmartYieldPoolViews(classes, _syDataState[0], protocols) : null}
+                {tabIndex === 1 ? GenerateSmartYieldPositions(t, _syDataState[1]) : null}
+            </Paper>
+        </CardContent>
+    )
+}
+
+function GenerateSmartYieldPositions(t: I18NFunction, syPortfolioData: SYPortoflioModelData) {
+    return (
+        <PortfolioBalance
+            total={syPortfolioData.seniorValue + syPortfolioData.juniorValue}
+            data={[
+                [t('plugin_barnbridge_sy_senior_balance'), syPortfolioData.seniorValue],
+                [t('plugin_barnbridge_sy_junior_balance'), syPortfolioData.juniorValue],
+            ]}
+        />
+    )
+}
+
+function GenerateSmartYieldPoolViews(classes: Record<string, string>, syData: SYPoolModelData, protocols: string[]) {
+    return (
+        <div className={classes.root}>
+            {protocols.map((name: string) => (
+                <SmartYieldPoolView protocolName={name} coins={syData[name]} />
```

Comment:
The `key` was lost here.

Reply:
Again not too sure what this means. Is there some scope issue where the key is invalid?

Reply:
I mean this [Lists and Keys](https://reactjs.org/docs/lists-and-keys.html#keys).

Reply:
Ah cool i never knew this exist -- will add the keys!